retry/lib/retry.dart

```diff
@@ -122,13 +123,22 @@ class RetryOptions {
     FutureOr<T> Function() fn, {
     FutureOr<bool> Function(Exception) retryIf,
     FutureOr<void> Function(Exception) onRetry,
+    FutureOr<bool> Function(dynamic) retryIfValue,
+    FutureOr<void> Function(dynamic) onRetryIfValue,
   }) async {
     int attempt = 0;
     // ignore: literal_only_boolean_expressions
     while (true) {
       attempt++; // first invocation is the first attempt
       try {
-        return await fn();
+        var value = await fn();
+        if (retryIfValue == null || !(await retryIfValue(value))) {
```

Comment:
```suggestion
        if (attempt >= maxAttempts || retryIfValue == null || !(await retryIfValue(value))) {
```

Shouldn't there also be a `maxAttempts` check?
Otherwise, the users would have to count by themselves to not exceed the attempts if the value/result repeatedly triggers a retry. (e.g. HTTP 503 for longer than 30 seconds)